tests/test_taar_similarity.py

```diff
@@ -288,15 +327,15 @@ def test_compute_donors(spark, addon_whitelist, multi_clusters_df):
     # Perform the clustering on our test data. We expect
     # 3 clusters out of this and 10 donors.
     _, donors_df = taar_similarity.get_donors(
-        spark, 3, 10, addon_whitelist, random_seed=42
+        spark, 3, 10, addon_whitelist, date_from='20180101', random_seed=42
     )
     donors = taar_similarity.format_donors_dictionary(donors_df)
 
     # Even if we requested 10 donors, it doesn't mean we will receive
     # precisely that number. All we can do is check that the number of
     # donors should always be >= 2 * num_clusters. Since we're fixing
-    # the seed for this test, we can just assert that we receive 14 donors.
-    assert len(donors) == 14
+    # the seed for this test, we can just assert that we receive 25 donors.
+    assert len(donors) == 25
```

Comment:
essentially this is providing test coverage for the expected behavior of `BisectingKMeans` over our data space for a sample of client telemetry data. Short of generating a more realistic sample of clients inside of the test case, I'm not sure what would constitute satisfactory test coverage here. 

I am comfortable with supposing that the built-in pyspark.ml.clustering functions are correctly implemented.
R+ from me, but I'm happy to have an issue opened if anyone has a specific strategy for meaningful test coverage in mind.

@akkomar @Dexterp37 can we merge it in, please?

Reply:
@mlopatka sorry, I wasn't clear here - I was concerned about the change I needed to do here (14->25), not the test itself. The only explanation I can think of is the pipeline being sensitive to ordering of input data (as this might have changed because of grouping clients and filtering out duplicates).

Reply:
Ah, I see what you mean. I do not think that we need to be concerned (from a testing perspective) with the increase in number of clusters. 

I suspect the culprit here is indeed a change in order that was a consequence from the change in the schema to adhere to clients_daily. The random seed being held constant but with a new set of client profiles being generated for testing may cause this change. The client_id's should not have any effect on the clustering algorithm itself. 

One possibility that we could explore to dig a tiny bit deeper is to sort `donors_df` by client_id before the call of: `donors = taar_similarity.format_donors_dictionary(donors_df)` to ensure that we consistently get a specific number of clusters to specify int he test.

Reply:
@mlopatka good idea - I tried sorting `users_sample` in `get_donors` before and after the change, it makes number of donors stable. I'm not incorporating this change here as that would require more refactoring just to expose sorting feature just for tests, we can revisit it later.
I can merge it after you make approving review.

Reply:
r+ from me, but I am not an approved reviewer for this repo.
we need either @Dexterp37 @acmiyaguchi  or @mreid-moz to do the formal thing.